File: code-generator/src/main/java/io/github/potjerodekool/openapi/log/Logger.java
package io.github.potjerodekool.openapi.log;

import org.checkerframework.checker.nullness.qual.Nullable;

/**
 * A logger interface to allow logging to the log of the buildsystems.
 */
public interface Logger {

    static Logger getLogger(final String name) {
        return LoggerFactory.getLogger(name);
    }

    default void info(final String message) {
        log(LogLevel.INFO, message);
    }

    default void log(final LogLevel level, final @Nullable String message) {
        this.log(level, message, null);
    }

    void log(LogLevel level, @Nullable String message, @Nullable Throwable exception);

}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/log/Logger.kt
package io.github.potjerodekool.openapi.common.log

/**
 * A logger interface to allow logging to the log of the buildsystems.
 */
interface Logger {
    fun info(message: String?) {
        log(LogLevel.INFO, message)
    }

    fun log(level: LogLevel?, message: String?) {
        this.log(level, message, null)
    }

    fun log(level: LogLevel?, message: String?, exception: Throwable?)

    companion object {
        fun getLogger(name: String): Logger {
            return LoggerFactory.getLogger(name)
        }
    }
}
